packages/http-client-java/eng/scripts/Test-Packages.ps1

```diff
@@ -20,22 +20,44 @@ Invoke-LoggedCommand "mvn -version"
 Push-Location $packageRoot
 try {
     if ($UnitTests) {
-        Push-Location "$packageRoot"
-        try {
-            Write-Host "Current PATH: $env:PATH"
-            Write-Host "Current JAVA_HOME: $Env:JAVA_HOME"
-            $env:JAVA_HOME = $env:JAVA_HOME_21_X64
-            Write-Host "Updated JAVA_HOME: $Env:JAVA_HOME"
+        Write-Host "Current PATH: $env:PATH"
+        Write-Host "Current JAVA_HOME: $Env:JAVA_HOME"
+        $env:JAVA_HOME = $env:JAVA_HOME_21_X64
+        Write-Host "Updated JAVA_HOME: $Env:JAVA_HOME"
 
-            $env:PATH = "$env:JAVA_HOME\bin;$env:PATH"
+        $env:PATH = "$env:JAVA_HOME\bin;$env:PATH"
 
-            Write-Host "Updated PATH: $env:PATH"
-            # test the emitter
-            Invoke-LoggedCommand "npm run build" -GroupOutput
-            
+        Write-Host "Updated PATH: $env:PATH"
+        
+        # cadl-ranch tests (unit tests included in java/typescript package build)
```

Comment:
It seems odd that you're moving generation checks out of the `if ($GenerationChecks) {` block

Reply:
Here the code only runs the end2end tests in cadl-ranch.

The reason is that the release pipeline has a step to upload the cadl-ranch test report, which requires we run these tests.
https://github.com/microsoft/typespec/blob/36d2930af1b86ff2df407a008d80fd99fcb19e36/eng/emitters/pipelines/templates/steps/test-step.yml#L69-L78

Let me know if there is some alternatives. E.g. can we do `EnableCadlRanchReport=false` in release? Or add `-GenerationChecks` [here](https://github.com/microsoft/typespec/blob/36d2930af1b86ff2df407a008d80fd99fcb19e36/packages/http-client-java/eng/pipeline/publish.yml#L28)?